libraries/AP_HAL_Linux/Util.h

```diff
@@ -48,7 +48,7 @@ class Util : public AP_HAL::Util {
      */
     void set_system_clock(uint64_t time_utc_usec);
     const char* get_custom_log_directory() { return custom_log_directory; }
-    const char* get_custom_terrain_directory() { return custom_terrain_directory; }
+    const char* get_custom_terrain_directory() const final { return custom_terrain_directory; }
```

Comment:
```c++
class A {
public:
	virtual void foo() { }
};

class B : public A {
public:
	void fooo() final { }
};

int main() {
	return 0;
}
```

```bash
g++ -std=gnu++11 -o /tmp/a /tmp/a.cpp
/tmp/a.cpp:8:7: error: ‘void B::fooo()’ marked ‘final’, but is not virtual
  void fooo() final { }
       ^~~~
```

As said above.   Since the message is not so clear and the user could be tempted to add a virtual do the dervided method, I added the override.